editor/doc/http-api.md

```diff
@@ -0,0 +1,98 @@
+  "show-build-errors"  : "Show the Build Errors tab.",
+  "show-console"       : "Show the Console tab.",
+  "show-curve-editor"  : "Show the Curve Editor tab.",
+  "support-forum"      : "Open the Defold Support Forum in a web browser.",
+  "toggle-pane-bottom" : "Toggle visibility of the bottom editor pane.",
+  "toggle-pane-left"   : "Toggle visibility of the left editor pane.",
+  "toggle-pane-right"  : "Toggle visibility of the right editor pane."
+}
+```
+
+#### Response Codes
+Commands are fire-and-forget. You will not know if a command succeeds or not, only if your request was accepted. As such, all command requests will return the same response codes. They are as follows:
+
+* `202 Accepted` - The request was accepted and we are processing it. We use this instead of `200 OK` since we won't be able to tell you when it is done or how it went.
+* `400 Bad Request` - The request was malformed.
+* `404 Not Found` - The string after `command/` is not a supported command.
+* `405 Method Not Allowed` - The command is supported, but it can't be executed right now (for example if a build is already in progress).
+* `500 Internal Server Error` - Something unexpected went wrong when executing the command. The error will be logged in the editor log.
+
+### The Console Endpoint
+The console endpoint allows external tools to obtain the current contents of the Console View in the editor. Send a GET request to `http://localhost:[port]/console` to obtain the console contents in JSON format. The returned object contains `lines` and `regions`.
+
+* The `lines` are returned as an array of strings, each string representing a line in the Console.
+* The `regions` contain metadata about sub-regions of the `lines` that can inform formatting, etc.
+
+#### Region Objects
+Each region object will be in the following format:
+```json
+{
+  "from": {"row": 0, "col": 0},
+  "to":   {"row": 0, "col": 42},
+  "type": "repeat"
+}
+```
+* `from` and `to` specify the row and columns that encompass the region as zero-based indices into the `lines`.
+* `type` is the type of the region, and can inform formatting.
+
+#### Region Types
+Depending on the `type` of a region, it might have additional fields. Here is a non-exhaustive list of region types used in the editor.
+
+* `eval-expression` - An expression posted from the Evaluate Lua field in the debugger.
+* `eval-result` - The result returned by the `eval-expression`.
+* `eval-error` - A runtime error resulting from an `eval-expression`.
+* `extension-output` - Output from an editor script or extension.
+* `extension-error` - Errors from an editor script or extension.
+* `repeat` - Repeated console output that has been collapsed. Its `count` is the number of repeats.
+* `resource-reference` - A substring that matches an existing path in the project. It has a `proj-path-candidates` array of project-relative paths to matching resources, and can also have a zero-based `row` index into the resource if it is text-based.
```

Comment:
Maybe we should rename the key to `project-path-candidates` to make it clearer?

Reply:
I'm torn about this. The term `proj-path` means a very specific thing in all places. It denotes a project-relative path with a leading slash, I.e. `/main/main.collection`. I think we should use the term in our APIs as well since it is useful and ubiquitous in the editor.